### PR TITLE
API and front end setup. Deployed to firebase.

### PR DESCRIPTION
**Pull request by dsiy**

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -1,24 +1,20 @@
-import React from 'react';
-import logo from './logo.svg';
+import React, { useState, useEffect } from 'react';
 import './App.css';
+import { List, Typography, Divider } from 'antd';
 
 function App() {
+  const [requests, setRequests] = useState([])
```
Comment (tekkeon) [inline client/src/App.js]: Add semicolon and add new line after

**Review by twtrinh — COMMENTED** [review]
Mitch said you have to reformat this, please do this immediately.

File: client/src/App.js [hunk 2]
```diff
@@ -1,24 +1,20 @@
...
+  useEffect(() => {
+    fetch("/api")
+      .then(res => res.json())
+      .then(req => setRequests(req.map((req, index)=>({...req, index}))))
+  }, [])
```
Comment (tekkeon) [inline client/src/App.js]: Add new line after this

File: client/src/App.js [hunk 3]
```diff
@@ -1,24 +1,20 @@
...
+import React, { useState, useEffect } from 'react';
 import './App.css';
+import { List, Typography, Divider } from 'antd';
 
 function App() {
+  const [requests, setRequests] = useState([])
+  useEffect(() => {
+    fetch("/api")
+      .then(res => res.json())
+      .then(req => setRequests(req.map((req, index)=>({...req, index}))))
```
Comment (tekkeon) [inline client/src/App.js]: Add semicolon

File: client/src/App.js [hunk 4]
```diff
@@ -1,24 +1,20 @@
...
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+    <div>
+      <List bordered={true} 
+        dataSource={requests}
+        renderItem={request => <List.Item><p>{request.index + " " + request.name}</p></List.Item>}
```
Comment (tekkeon) [inline client/src/App.js]: New lines for the JSX elements

File: functions/index.js [hunk 5]
```diff
@@ -1,7 +1,54 @@
 const functions = require('firebase-functions')
 const express = require('express')
 const app = express()
-app.get('/', (req, res) => {
-  res.send('hello world!')
+const bodyParser = require('body-parser')
```
Comment (tekkeon) [inline functions/index.js]: Add lots of semicolons

File: functions/index.js [hunk 6]
```diff
@@ -1,7 +1,54 @@
...
+const admin = require('firebase-admin')
+const cors = require('cors')
+
+admin.initializeApp(functions.config().firebase);
```
Comment (tekkeon) [inline functions/index.js:8]: New line after this

File: functions/index.js [hunk 7]
```diff
@@ -1,7 +1,54 @@
...
+app.use(cors({ origin: true }));
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: false }));
```
Comment (tekkeon) [inline functions/index.js:11]: New line after

File: functions/index.js [hunk 8]
```diff
@@ -1,7 +1,54 @@
...
+      res.status(201).send(`Created a new user: ${newDoc.id}`);
+    } else {
+      functions.logger.log("already added in queue");
+      res.status(400).send('already added in queue!!');
+    }
+
+  } catch (error) {
+    functions.logger.error(error)
+    res.status(400).send(`User should cointain name, pantherID, course, and requested time!!!`)
+  }
```
Comment (tekkeon) [inline functions/index.js:45]: New line after this

File: functions/index.js [hunk 9]
```diff
@@ -1,7 +1,54 @@
...
 })
-exports.app = functions.https.onRequest(app)
\ No newline at end of file
+app.get('/api', async (req, res) => {
+  functions.logger.log("balls")
+  try{
```
Comment (tekkeon) [inline functions/index.js]: Add a space after try and a space before and after the catch stuff

File: functions/index.js [hunk 10]
```diff
@@ -1,7 +1,54 @@
...
+
+admin.initializeApp(functions.config().firebase);
+app.use(cors({ origin: true }));
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: false }));
+const db = admin.firestore();
+const requestCollectionName = 'tutoringRequests';
+
+app.post('/api/requests', async (req, res) => {
+  // functions.logger.log(req)
```
Comment (tekkeon) [inline functions/index.js]: Remove comment

File: functions/index.js [hunk 11]
```diff
@@ -1,7 +1,54 @@
...
+  try {
+    const request = {
+      name: req.body.name,
+      pantherID: req.body.pantherID,
+      course: req.body.course,
+      completed: false,
+      requestedAt: new Date().toString()
+    }
+    const requestCollection = db.collection(requestCollectionName);
+    const querySnapshot = await requestCollection.where("pantherID", "==", request.pantherID).where("course", "==", request.course ).get()
```
Comment (tekkeon) [inline functions/index.js]: Put the query and the add calls to the db in separate try/catch blocks. That way if something goes wrong with one of the calls, we can have different logs for each call in the catch blocks.

File: functions/index.js [hunk 12]
```diff
@@ -1,7 +1,54 @@
...
+
+  } catch (error) {
+    functions.logger.error(error)
+    res.status(400).send(`User should cointain name, pantherID, course, and requested time!!!`)
+  }
 })
-exports.app = functions.https.onRequest(app)
\ No newline at end of file
+app.get('/api', async (req, res) => {
+  functions.logger.log("balls")
```
Comment (tekkeon) [inline functions/index.js]: ...

File: functions/index.js [hunk 13]
```diff
@@ -1,7 +1,54 @@
...
+  try{
+    const testGet = (await db.collection(requestCollectionName).get()).docs.map(doc => doc.data());
+    functions.logger.log(testGet);
+    res.send(JSON.stringify(testGet))
+  }catch (error){
+    functions.logger.error(error);
+    res.send("my god. you are so disappointing. i wish you were never born.")
```
Comment (tekkeon) [inline functions/index.js]: ...

File: functions/index.js [hunk 14]
```diff
@@ -1,7 +1,54 @@
...
+  }
+
```
Comment (tekkeon) [inline functions/index.js]: Remove blank line

Comment (dsiy) [reply functions/index.js]: Removed blank line on 52

File: functions/index.js (hunk 13 above)
Comment (dsiy) [reply functions/index.js]: Some monkey wrote that.

File: functions/index.js (hunk 12 above)
Comment (dsiy) [reply functions/index.js]: Reference: basket, volley, base, soccer, tennis, or base

File: functions/index.js (hunk 11 above)
Comment (dsiy) [reply functions/index.js]: Believe that I changed this to what you intended. Please let me know if this is not the case.

File: client/src/App.js (hunk 4 above)
Comment (dsiy) [reply client/src/App.js]: Not sure if this is what you intended? Please let me know

File: functions/index.js [hunk 15]
```diff
@@ -23,8 +22,12 @@ app.post('/api/requests', async (req, res) => {
       requestedAt: new Date().toString()
     }
     const requestCollection = db.collection(requestCollectionName);
-    const querySnapshot = await requestCollection.where("pantherID", "==", request.pantherID).where("course", "==", request.course ).get()
-
+    const querySnapshot = await requestCollection.where("pantherID", "==", request.pantherID).where("course", "==", request.course ).get();
```
Comment (tekkeon) [inline functions/index.js]: Scope issue here. If you declare the variable inside this try block, you can't access it inside the lower try block. You'll have to create a var before the two try blocks (set it equal to null) and modify it in the first one then you can access it in the second one.

File: functions/index.js [hunk 16]
```diff
@@ -23,8 +22,12 @@ app.post('/api/requests', async (req, res) => {
...
+  }catch (error){
+    functions.logger.error(error);
+    res.status(400).send("Issue with query");
+  }
```
Comment (tekkeon) [inline functions/index.js:30]: New line

File: functions/index.js [hunk 17]
```diff
@@ -23,8 +22,12 @@ app.post('/api/requests', async (req, res) => {
       requestedAt: new Date().toString()
     }
     const requestCollection = db.collection(requestCollectionName);
-    const querySnapshot = await requestCollection.where("pantherID", "==", request.pantherID).where("course", "==", request.course ).get()
-
+    const querySnapshot = await requestCollection.where("pantherID", "==", request.pantherID).where("course", "==", request.course ).get();
+  }catch (error){
```
Comment (tekkeon) [inline functions/index.js]: Spaces before catch and after )

File: functions/index.js [hunk 18]
```diff
@@ -13,8 +13,7 @@ const db = admin.firestore();
 const requestCollectionName = 'tutoringRequests';
 
 app.post('/api/requests', async (req, res) => {
-  // functions.logger.log(req)
-  try {
+  try{
```
Comment (tekkeon) [inline functions/index.js]: Space after try

File: functions/index.js [hunk 19]
```diff
@@ -35,20 +38,19 @@ app.post('/api/requests', async (req, res) => {
...
 })
 app.get('/api', async (req, res) => {
-  functions.logger.log("balls")
+  functions.logger.log(req);
   try{
     const testGet = (await db.collection(requestCollectionName).get()).docs.map(doc => doc.data());
     functions.logger.log(testGet);
-    res.send(JSON.stringify(testGet))
+    res.send(JSON.stringify(testGet));
   }catch (error){
```
Comment (tekkeon) [inline functions/index.js]: Same catch spacing thing

**Review by tekkeon — APPROVED** [review]

**Pull request merged**